### PR TITLE
(PUP-3254) Removing Modulefile functionality

### PR DESCRIPTION
**Pull request by andersonmills**
Before this commit, the use of Modulefile for containing the metadata
for a module was deprecated, but still usable. This commit removes that
functionality completely, removes the deprecation warnings for
Modulefile, removes the spec and acceptance tests for Modulefile, and
issues a warning that Modulefile will be ignored as a special file.

### CODE REVIEW

**Comment by joshcooper** [comment]
@andersonmills looks like a typo in the commit summary

**Comment by andersonmills** [comment]
s/funcitonality/functionality

File: acceptance/tests/modules/changes/missing_metadata_json.rb [hunk 1]
```diff
@@ -1,22 +0,0 @@
-test_name 'puppet module changes (on a module which is missing metadata.json)'
```
Comment (adreyer) [inline acceptance/tests/modules/changes/missing_metadata_json.rb]: I think this test is still valid we should just get rid of the modulefile.

File: lib/puppet/face/module/generate.rb [hunk 2]
```diff
@@ -107,9 +107,9 @@
       begin
         # A default dependency for all newly generated modules is being
         # introduced as a substitute for the comments we used to include in the
-        # Modulefile. While introducing a default dependency is less than
-        # perfectly desirable, the cost is low, and the syntax is obtuse enough
-        # to justify its inclusion.
+        # previous module data specifactions. While introducing a default
```
Comment (adreyer) [inline lib/puppet/face/module/generate.rb]: specifaction -> specification

**Comment by andersonmills** [comment]
@adreyer I've updated the acceptance tests as @cowofevil, you, and I planned.

**Comment by adreyer** [comment]
+1

**Pull request merged**